mod/gc_mobile_api/models/opportunity.php

```diff
@@ -598,3 +653,360 @@ function accept_post($user, $guid, $lang)
+	$_SESSION['mission_duplicating_override_first'] = true;
+	unset($_SESSION['mission_uncheck_post_mission_disclaimer']);
+	
+	$err = '';
+	
+	$first_form = elgg_get_sticky_values('firstfill');
+
+	$first_form['name'] = $formData["name"];
+	$first_form['department'] = $formData["departement"];
+	$first_form['email'] = $formData["email"];
+	$first_form['phone'] = $formData["phone"];
+	$first_form['disclaimer'] = $formData["agree"];
+
+	// Error checking function.
+$err .= mm_first_post_error_check($first_form);
+	if ($err == '') {
+		$_SESSION['missions_pass_department_to_second_form'] = mo_get_last_input_node($first_form);
+		forward(elgg_get_site_url() . 'missions/mission-post/step-two');
+	} else {
+		error_log('error in first form: '.$err);
+	}
+
+
+	return $formData["name"];
+}
+
+
+function create_opportinities2($user, $formData, $lang)
+{
+	$user_entity = is_numeric($user) ? get_user($user) : (strpos($user, '@') !== false ? get_user_by_email($user)[0] : get_user_by_username($user));
+	if (!$user_entity) {
+		return "User was not found. Please try a different GUID, username, or email address";
+	}
+	if (!$user_entity instanceof ElggUser) {
+		return "Invalid user. Please try a different GUID, username, or email address";
+	}
+
+	if (!elgg_is_logged_in()) {
+		login($user_entity);
+	}
+
+	elgg_make_sticky_form('secondfill');
+	$_SESSION['mission_duplicating_override_second'] = true;
+	
+	$err = '';
+	$first_form = elgg_get_sticky_values('firstfill');
+	$second_form = elgg_get_sticky_values('secondfill');
+
+	$second_form['job_title'] = $formData["name"];
+	$second_form['role_type'] = $formData["offert"];
+	$second_form['job_type'] = $formData["type"];
+	$second_form['job_area'] = $formData["program"];
+	$second_form['number'] = $formData["num_opt"];
+	$second_form['start_date'] = $formData["start_date"];
+	$second_form['completion_date'] = $formData["completion_date"];
+	$second_form['deadline'] = $formData["deadline"];
+	$second_form['description'] = $formData["description"];
+	
+	// Error checking function.
+	$err .= mm_first_post_error_check($first_form);
+	$err .= mm_second_post_error_check($second_form);
+	
+	if ($err == '') {
+		forward(elgg_get_site_url() . 'missions/mission-post/step-three');
+	} else {
+		error_log('error in first form: '.$err);
+	}
+	
+
+
+	error_log('first :'.print_r($first_form,true));
+	error_log('second :'.print_r($second_form,true));
+	return $formData["title"];
+}
+
+function create_opportinities3($user, $formData,$lang)
+{
+	$user_entity = is_numeric($user) ? get_user($user) : (strpos($user, '@') !== false ? get_user_by_email($user)[0] : get_user_by_username($user));
+	if (!$user_entity) {
+		return "User was not found. Please try a different GUID, username, or email address";
+	}
+	if (!$user_entity instanceof ElggUser) {
+		return "Invalid user. Please try a different GUID, username, or email address";
+	}
+
+	if (!elgg_is_logged_in()) {
+		login($user_entity);
+	}
+
+elgg_make_sticky_form('thirdfill');
```

Comment:
If I remove this function elgg_make_sticky_form, I'm not able to keep my value of all the form.
It work fine when I test on my localhost and chrome. Do you think this function will not work on phone?

Reply:
I don't think it would since the sessions variables that keep the form info are set on gccollab.ca and not the app itself. If you have your changes in the latest version of the app, we could test this. I'll upload this latest version of `opportunity.php` onto prod.

Reply:
Test on prod, and with the elgg_make_sticky_form, it work fine